plugin/src/test/java/io/jenkins/plugins/analysis/warnings/steps/StepsOnAgentITest.java

```diff
@@ -99,4 +100,28 @@ void shouldRecordOutputOfParallelSteps() {
         assertThat(first.getResult().getIssues()).hasSize(5);
         assertThat(second.getResult().getIssues()).hasSize(3);
     }
+
+    /**
+     * Verifies that source files are not retained in the Jenkins build folder when
+     * the 'sourceCodeRetention' policy is set to 'NEVER'
+     **/
+    @Test
+    void shouldNotCopySourcesWhenSourceCodeRetentionIsNever() {
+        Slave agent = createAgentWithEnabledSecurity("agent");
+
+        WorkflowJob project = createPipeline();
+
+        createFileInAgentWorkspace(agent, project, "Test.java", JAVA_CONTENT);
+
+        project.setDefinition(new CpsFlowDefinition("node('agent') {\n"
+                + "    echo '[javac] Test.java:39: warning: Test Warning'\n"
+                + "    recordIssues tool: java(), sourceCodeRetention: 'NEVER'\n"
+                + "}", true));
+
```

Comment:
Can you check here that the console log reports that no files are stored? Similar to 

```
assertThat(getConsoleLog(result)).contains("1 copied", "0 not in workspace", "0 not-found", "0 with I/O error");
```

there should be a log message that shows that source code files are not stored...

Reply:
thank you @KalleOlaviNiemitalo for the insights, @uhafner what do you think about this ?

Reply:
I think it would be better to add a logging statement in the else part so everybody can see that the parameter is correctly set. Something like "Skipping copying of affected files"